### PR TITLE
I212 - In an AllinsonFlex profile, edit property visibility (with an admin_only check)

### PR DESCRIPTION
**Pull request by ShanaLMoore**
Pulled over work done by IU: https://github.com/IU-Libraries-Joint-Development/essi/pull/344/commits

ref: https://gitlab.com/notch8/utk-hyku/-/issues/212

# Summary

This work was done by Indiana University. 

!! Possible breaking change: the app using Allinson Flex needs to define a admin? method

# Acceptance Criteria

- [ ] non admin users should not see properties that are marked as admin_only, in the metadata profile. 

# Screenshots or Video

**admin_only indexing was added to rights_statement**

AS A LOGGED OUT USER: 
Screen Shot 2022-10-13 at 4 39 47 PM

AS AN LOGGED IN ADMIN USER: 
Screen Shot 2022-10-13 at 4 03 31 PM

# Testing Instructions

1. add admin_only and displayable indexing to a property in the metadata yaml profile
2. create a work. fill in data for that admin_only property. 
- [ ] When a user is logged in as an admin, they should see that property on the work's show page
- [ ] When a user is logged out, they should not see that admin_only property

### CODE REVIEW

File: app/models/concerns/allinson_flex/dynamic_metadata_behavior.rb [hunk 1]
```diff
@@ -91,7 +91,8 @@ def dynamic_schema=(value)
     def dynamic_schema_service(args = {})
       # clear memoizaiton when as_id changes
       old_as_id = @as_id
-      @as_id = args[:as_id] || admin_set_id || AdminSet::DEFAULT_ID
+      
+      @as_id = args[:as_id] || try(:admin_set_id) || AdminSet::DEFAULT_ID
```
Comment (laritakr) [inline app/models/concerns/allinson_flex/dynamic_metadata_behavior.rb:95]: Is this change related to this admin_only change?

**Review by laritakr — APPROVED** [review]
Looks ok, but I as wondering about that one change.

File: app/models/concerns/allinson_flex/dynamic_metadata_behavior.rb (hunk 1 above)
Comment (ShanaLMoore) [reply app/models/concerns/allinson_flex/dynamic_metadata_behavior.rb]: removed
